src/main/java/uk/ac/sanger/sccp/utils/BasicUtils.java

```diff
@@ -247,4 +259,32 @@ public static String trimAndRequire(String text, String message) throws IllegalA
         }
         return text;
     }
+
+    /**
+     * Predicate for filtering out duplicates from a stream.
+     * This uses a simple hashset to track what was seen: it is not guaranteed
+     * to work for a parallel stream.
+     * The keys (the results of the supplied function) must be hashable.
+     * @param function The function to extract the keys from the objects in the stream.
+     * @param <T> The stream type.
+     * @param <V> The type of key used to distinguish objects in the stream.
+     * @return a predicate for filtering out duplicates.
+     */
+    public static <T, V> Predicate<T> distinctBySerial(Function<? super T, V> function) {
+        final Set<V> seen = new HashSet<>();
+        return x -> seen.add(function.apply(x));
+    }
+
+    /**
+     * Predicate for filtering out duplicates in a string case insensitively.
+     * This uses a simple hashset to track what was seen: it is not guaranteed
+     * to work for a parallel stream.
+     * Case insensitivity for the purposes of this method means that the strings are equal
+     * when converted to upper case.
+     * @return a predicate for filtering out duplicate strings, case insensitively.
+     */
+    public static Predicate<String> distinctUCSerial() {
```

Comment:
These look good. Where are they used?

Reply:
Not used. I brought them in and then I rewrote the code I was using them in to work another way.